lib/cartopy/crs.py

```diff
@@ -1450,6 +1450,55 @@ def y_limits(self):
         return self._ylim
 
 
+class AzimuthalEquidistant(Projection):
+    """
+    An Azimuthal Equidistant projection
+
+    This projection provides accurate angles about and distances through the
+    central position. Other angles, distances, or areas may be distorted.
+    """
+
+    def __init__(self, central_longitude=0.0, central_latitude=0.0,
+                 false_easting=0.0, false_northing=0.0,
+                 globe=None):
+        """
+        Kwargs:
+
+            * central_longitude - The true longitude of the central meridian in
+                                  degrees. Defaults to 0.
+            * central_latitude - The true latitude of the planar origin in
+                                 degrees. Defaults to 0.
+            * false_easting - X offset from the planar origin in metres.
+                              Defaults to 0.
+            * false_northing - Y offset from the planar origin in metres.
+                               Defaults to 0.
+            * globe - An instance of :class:`cartopy.crs.Globe`. If omitted, a
+                      default globe is created.
+
+        """
+        proj4_params = [('proj', 'aeqd'), ('lon_0', central_longitude),
+                        ('lat_0', central_latitude),
+                        ('x_0', false_easting), ('y_0', false_northing)]
+        super(AzimuthalEquidistant, self).__init__(proj4_params, globe=globe)
+        self._max = 2e7
```

Comment:
I tried the following:
```
import matplotlib.pyplot as plt
import cartopy.crs as ccrs

globe = ccrs.Globe(semimajor_axis=1000, semiminor_axis=500,ellipse=None)
aeqd = ccrs.AzimuthalEquidistant(globe=globe)

plt.figure()
ax = plt.axes(projection=aeqd)
ax.coastlines(resolution='110m')
ax.gridlines()
plt.show()
```
It takes a while, but eventually fails with:
```
Traceback (most recent call last):
  File ".../matplotlib/backends/backend_qt4.py", line 299, in resizeEvent
    self.draw()
  File ".../matplotlib/backends/backend_qt4agg.py", line 154, in draw
    FigureCanvasAgg.draw(self)
  File ".../matplotlib/backends/backend_agg.py", line 451, in draw
    self.figure.draw(self.renderer)
  File ".../matplotlib/artist.py", line 55, in draw_wrapper
    draw(artist, renderer, *args, **kwargs)
  File ".../matplotlib/figure.py", line 1034, in draw
    func(*args)
  File ".../matplotlib/artist.py", line 55, in draw_wrapper
    draw(artist, renderer, *args, **kwargs)
  File ".../cartopy/mpl/geoaxes.py", line 358, in draw
    inframe=inframe)
  File ".../matplotlib/artist.py", line 55, in draw_wrapper
    draw(artist, renderer, *args, **kwargs)
  File ".../matplotlib/axes.py", line 2086, in draw
    a.draw(renderer)
  File ".../matplotlib/artist.py", line 55, in draw_wrapper
    draw(artist, renderer, *args, **kwargs)
  File ".../cartopy/mpl/feature_artist.py", line 103, in draw
    extent = ax.get_extent(feature_crs)
  File ".../cartopy/mpl/geoaxes.py", line 500, in get_extent
    p = self._get_extent_geom(crs)
  File ".../cartopy/mpl/geoaxes.py", line 546, in _get_extent_geom
    self.projection)
  File ".../cartopy/crs.py", line 172, in project_geometry
    return getattr(self, method_name)(geometry, src_crs)
  File ".../cartopy/crs.py", line 306, in _project_polygon
    rings.extend(self._attach_lines_to_boundary(multi_lines, is_ccw))
  File ".../cartopy/crs.py", line 425, in _attach_lines_to_boundary
    raise RuntimeError('Unidentified problem with '
RuntimeError: Unidentified problem with geometry, linestring being re-added
```
and the plot is empty. That being said, I _also_ tried this out with `Orthographic`, and the result is a _very_ zoomed out figure, so I guess there's a bug there too. I will investigate if I can determine the boundary in a bit more generic manner.